### PR TITLE
codegen: Fix whitelisting inside namespaces.

### LINKED ISSUE #287: Namespace mods are not generated correctly when whitelist-type is specified
For file:
```c++
namespace outer {
  namespace inner {
    struct Helper {};
  }
  struct Test {
    inner::Helper helper;
  };
}
```

`cargo run -- test.hpp --enable-cxx-namespaces` generates:
```rust
pub mod root {
    use root;
    pub mod outer {
        use root;
        pub mod inner {
            use root;
            #[repr(C)]
            #[derive(Debug, Copy)]
            pub struct Helper {
                pub _address: u8,
            }
            #[test]
            fn bindgen_test_layout_Helper() {
                assert_eq!(::std::mem::size_of::<Helper>() , 1usize);
                assert_eq!(::std::mem::align_of::<Helper>() , 1usize);
            }
            impl Clone for Helper {
                fn clone(&self) -> Self { *self }
            }
        }
        #[repr(C)]
        #[derive(Debug, Copy)]
        pub struct Test {
            pub helper: root::outer::inner::Helper,
        }
        #[test]
        fn bindgen_test_layout_Test() {
            assert_eq!(::std::mem::size_of::<Test>() , 1usize);
            assert_eq!(::std::mem::align_of::<Test>() , 1usize);
        }
        impl Clone for Test {
            fn clone(&self) -> Self { *self }
        }
    }
}
```

but `cargo run -- test.hpp --enable-cxx-namespaces --whitelist-type '.*'` generates:
```rust
#[repr(C)]
#[derive(Debug, Copy)]
pub struct Helper {
    pub _address: u8,
}
#[test]
fn bindgen_test_layout_Helper() {
    assert_eq!(::std::mem::size_of::<Helper>() , 1usize);
    assert_eq!(::std::mem::align_of::<Helper>() , 1usize);
}
impl Clone for Helper {
    fn clone(&self) -> Self { *self }
}
#[repr(C)]
#[derive(Debug, Copy)]
pub struct Test {
    pub helper: root::outer::inner::Helper,
}
#[test]
fn bindgen_test_layout_Test() {
    assert_eq!(::std::mem::size_of::<Test>() , 1usize);
    assert_eq!(::std::mem::align_of::<Test>() , 1usize);
}
impl Clone for Test {
    fn clone(&self) -> Self { *self }
}
```
which won't build because it tries to use an undefined module.

cc @fitzgen @emilio

### PR DESCRIPTION
**Pull request by emilio**
fixes #287 

r? @fitzgen

### CODE REVIEW

File: libbindgen/tests/headers/whitelist-namespaces-basic.hpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// bindgen-flags: --enable-cxx-namespaces --whitelist-type '.*Helper'
+
+namespace outer {
+  namespace inner {
+    struct Helper {};
+  }
+  struct Test {
```
Comment (upsuper) [inline libbindgen/tests/headers/whitelist-namespaces-basic.hpp]: If it is only whitelisting `.*Helper`, why is `Test` also generated?

Comment (emilio) [inline libbindgen/tests/headers/whitelist-namespaces-basic.hpp]: It isn't, though, right? Note there are two expectations file, one being your test case, and other white listing Helper.

Comment (upsuper) [inline libbindgen/tests/headers/whitelist-namespaces-basic.hpp]: Ahh... Right, it isn't... This is the header file not the generated one...

**Pull request merged**